### PR TITLE
Added documentation for more advanced configuration

### PR DESCRIPTION
**Pull request by tortilaman**
**- Summary**

I’ve added configuration documentation for the following
* Registering an Oauth application
* Configuring an external github Oauth authentication server ([Vencax’s](https://github.com/vencax/netlify-cms-github-oauth-provider))
* Using Github Enterprise Repositories
* Self-hosting Netlify CMS.

I finally got this stuff figured out on my install, and figured it would be nice to have it added to the documentation. As I continue getting others on-board I'll probably add more. It’s by no means comprehensive, but I think it’s a start. Looking
forward to hearing your thoughts.

**- Description for the changelog**

New documentation added.

**- A picture of a cute animal (not mandatory but encouraged)**
OMG I'm batman

### CODE REVIEW

File: docs/advanced-config.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  branch: master
+  api_root: https://github.some.domain.com/api/v3
+  site_id: static.site.url.com
+  base_url: https://auth.server.url.com
+```
+
+#### Collections
+You can find info on collections and translating your static site yaml front-matter to cms fields in the [quick-start](https://www.netlifycms.org/docs/quick-start/) section.
+
+### Continuous Delivery
```
Comment (erquhart) [inline docs/advanced-config.md]: This section is out of scope, not really CMS related.

File: docs/advanced-config.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+There are many ways to add Netlify CMS to your static site. This is by no means a comprehensive guide, but some of the more popular use-cases will be covered, including:
+
+* Configuration for a self-hosted Netlify CMS.
+* Configuring a third-party Github Oauth provider.
+* Using Github Enterprise repositories.
+
+If there's any other use-cases you're interested in or information you think is missing, you can file a [feature request](https://github.com/netlify/netlify-cms/issues), or [help us add it](https://github.com/netlify/netlify-cms/blob/master/CONTRIBUTING.md). Let's get started!
+
+## Register a Github Oauth App
+Information is available on the [Github Developer Documentation](https://developer.github.com/apps/building-integrations/setting-up-and-registering-oauth-apps/registering-oauth-apps/). Fill out the fields however you like, however **it is critical that your _authorization callback URL_ is correct**. This tells Github where to send a user after they've authenticated, and should match the URL of your Oauth provider. Use `https://your.app.com/callback` with Vencax's Oauth provider explained below.
```
Comment (erquhart) [inline docs/advanced-config.md]: We'll want to avoid centering project documentation around individual contributors (in this case Vencax). His (stellar!) contribution is the only one currently, but we're hoping to see more. A direction mention of his repo should happen just once, and in some sort of list that can be added to. Would you mind making that adjustment?

**Comment by erquhart** [comment]
@tortilaman thanks for this!! Great improvement that will help a lot of folks! Just a couple of changes requested - looking forward to merging :)

File: docs/advanced-config.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+Another convenient feature is the ability to have a different backend configuration depending on if you're running a development or production environment. This can be done like so:
+
+```yaml
+backend:
+  name: netlify-auth
+  repo: user/repository
+  branch: dev
+
+production:
```
Comment (erquhart) [inline docs/advanced-config.md]: I don't believe this works - let me know if I'm missing something. The CMS doesn't look for or process a `production` key from the config.

**Comment by erquhart** [comment]
[@tortilaman](https://github.com/tortilaman) thanks for this!! Great improvement that will help a lot of folks! Just a couple of changes requested - looking forward to merging :)

File: docs/advanced-config.md (hunk 2 above)
Comment (tortilaman) [inline docs/advanced-config.md]: Sure, I can reformat to accommodate that. I think there's an issue with checking whether or not a user has access to the particular repo anyways, so I'll have to look into that.

File: docs/advanced-config.md (hunk 3 above)
Comment (tortilaman) [inline docs/advanced-config.md]: Must be a leftover from either a particular boilerplate or a previous version then. I'll leave it out for now until I know better.

File: docs/advanced-config.md [hunk 4]
```diff
@@ -3,29 +3,25 @@
...
 
-After clicking "Register Application", you'll be provided with a `client_id` and `client_secret` on the next page that you'll use when configuring your Oauth provider
+After clicking "Register Application", you'll be provided with a `client_id` and `client_secret` on the next page that you'll use when configuring your Oauth Server
 
-## Configure an External Oauth Provider
+## Configure an External Oauth Server
+Netlify CMS is meant to be an open platform, so we're always looking to expand the ecosystem and find new ways to use it. Below is a list of currently submitted Oauth providers, but again feel free to [submit a pull request](https://github.com/netlify/netlify-cms/blob/master/CONTRIBUTING.md) if you think this list is missing anything.
```
Comment (erquhart) [inline docs/advanced-config.md]: ❤️

File: docs/advanced-config.md (hunk 1 above)
Comment (erquhart) [inline docs/advanced-config.md]: Should have been more clear - I meant the Continuous Delivery section is out of scope, not CMS related.

File: docs/advanced-config.md [hunk 5]
```diff
@@ -3,29 +3,25 @@
...
 
-Configuration is accomplished with environment variables, and is probably easiest with a `.env` file. An example would be:
-
-```
-NODE_ENV=production
-OAUTH_CLIENT_ID=f432a9casdff1e4b79c57
-OAUTH_CLIENT_SECRET=pampadympapampadympapampadympa
-GIT_HOSTNAME=https://github.some.domain.com
-```
-`GIT_HOSTNAME` is only necessary if you're using Github Enterprise. Additional config and install information for Vencax's Oauth provider is available in [its repository](https://github.com/vencax/netlify-cms-github-oauth-provider).
```
Comment (erquhart) [inline docs/advanced-config.md]: This was a great tip, we should keep it in. It will show up if anyone searches for GitHub enterprise in the docs.

Comment (tortilaman) [inline docs/advanced-config.md]: I've submitted a pull request to Vencax's repo that includes the Github Enterprise functionality, as well as documentation for it. Considering that all of those environment variables are specific to his repo, I figured I would remove it. I could add it back in some form though potentially.

From my PR on Vencax's repo:
```md
### Auth Provider Config

Configuration is done with environment variables, which can be supplied as command line arguments, added in your app  hosting interface, or loaded from a .env ([dotenv](https://github.com/motdotla/dotenv)) file.

**Example .env file:**

```
NODE_ENV=production
OAUTH_CLIENT_ID=f432a9casdff1e4b79c57
OAUTH_CLIENT_SECRET=pampadympapampadympapampadympa
REDIRECT_URL=https://your.server.com/callback
GIT_HOSTNAME=https://github.website.com
```

**Client ID & Client Secret:**
After registering your Oauth app, you will be able to get your client id and client secret on the next page.

**Redirect URL (optional):**
Include this if you  need your callback to be different from what is supplied in your Oauth app configuration.

**Git Hostname (Optional):**
This is only necessary for use with Github Enterprise.
```

**Comment by tortilaman** [comment]
@erquhart I might recommend mentioning that netlify-cms v0.4.3 or higher is required for custom authentication. My cms.js file was from v0.4.2, and had I known that was my problem figuring this out would've taken me minutes instead of days.

**Comment by erquhart** [comment]
@tortilaman very good point, thank you
